### ISSUE TITLE
Crashes when i tried to launch any game with hyperstone CPU

### ISSUE THREAD

**Issue by Panderner** [libretro/mame#235]
I've tried to launch any game with hyperstone CPU (Games from vamphalf, eolith and limenko) it crashes on boot including hynix variant (Mr. Dig, Crazy War and Mission Craft)

**Comment by tcamargo** [comment]
Could you please provide some logs?

**Comment by Panderner** [comment]
Ok here's a log:
[retroarch.log.zip](https://github.com/libretro/mame/files/6061766/retroarch.log.zip)

**Comment by tcamargo** [comment]
Just tested vamphalf and it's working.
Could you please test your rom on upstream mame?

**Comment by Panderner** [comment]
I'm using Android device I don't have a PC, standalone mame works fine.

**Comment by tcamargo** [comment]
Could you enable verbose and capture the log again? What. happens when you load the games? Does RA crash?

**Comment by Panderner** [comment]
Yes. Other games works fine but hyperstone CPUs crashes retroarch. I enabled logging verbosity and Here's a second log: 
[second log.zip](https://github.com/libretro/mame/files/6062301/second.log.zip)

**Comment by tcamargo** [comment]
Not much. Unfortunately, I don't have an android device to test. Game runs fine on Mac.

**Comment by Panderner** [comment]
> Unfortunately, I don't have an android device to test.

@tcamargo Can you buy any cheaper Android device?

**Comment by nemo93** [comment]
Hi. I do confirm same behavior on Raspberry Pi4 as well. Retroarch will segfault without further notice. I can't even get to the self-test/init screen. Tested with [Vamf x1/2](http://adb.arcadeitalia.net/dettaglio_mame.php?game_name=vamphalf) and [Legend of Heroes](http://adb.arcadeitalia.net/dettaglio_mame.php?game_name=legendoh) from a proper .230 romset. Both games do run fine with the FBNeo core - which relies on same roms.

I can provide a full log if requested. I'm copying just a few lines as follows, especially the very last one which reports the segfault:
```
Parameters:
Executing: /opt/retropie/emulators/retroarch/bin/retroarch -L /opt/retropie/libretrocores/lr-mame/mamearcade_
libretro.so --config /opt/retropie/configs/mame-libretro/retroarch.cfg "/home/pi/RetroPie/roms/mame-libretro/
vamphalf.7z" --verbose --appendconfig /dev/shm/retroarch.cfg
[INFO] RetroArch 1.8.8 (Git 9552f87)
[INFO] Redirecting save file to "/home/pi/RetroPie/roms/mame-libretro/vamphalf.srm".
[INFO] Redirecting save state to "/home/pi/RetroPie/roms/mame-libretro/vamphalf.state".
[INFO] === Build =======================================
[INFO] Capabilities:  NEON VFPv3 VFPv4
[INFO] Built: Feb 22 2021
[INFO] Version: 1.8.8
[INFO] Git: 9552f87
[INFO] =================================================
...
[snip]
...
[libretro INFO] ChangeAV: w:320 h:236 ra:1.333333.
/opt/retropie/supplementary/runcommand/runcommand.sh: line 1305:  7771 Segmentation fault /opt/retropie/emulators
/retroarch/bin/retroarch -L /opt/retropie/libretrocores/lr-mame/mamearcade_libretro.so --config 
/opt/retropie/configs/mame-libretro/retroarch.cfg "/home/pi/RetroPie/roms/mame-libretro/vamphalf.7z" --verbose
--appendconfig /dev/shm/retroarch.cfg
```

Raspberry Pi4 runs on a Retropie last version (4.7.12). Kernel 5.10.17-v7l+, 32-bit Raspberry OS and `fKMS` mode. Retroarch is 1.8.8 but the MAME core is up-to-date:

```
URL: https://github.com/libretro/mame.git
Branch: master
Commit: 98b0ba18a9109339c2ebf4f5945f6c3575301ba9
Date: 2021-04-12T17:53:19+02:00
```

**Comment by nemo93** [comment]
[Vamf x1/2](http://adb.arcadeitalia.net/dettaglio_mame.php?game_name=vamphalf) and [Legend of Heroes](http://adb.arcadeitalia.net/dettaglio_mame.php?game_name=legendoh) are working ok with **.257**. Yet on RPi the games are running too slow to be playable. They're both running fine on FB Neo.

@Panderner please confirm everything is fine on **.257** and if that's the case perhaps close that ticket off? Thanks!

**Comment by Panderner** [comment]
I tested on my POCO M3 with .257 it no longer crashes but it only run at 1 fps.
![Screenshot_2023-08-10-18-13-04-538_com retroarch aarch64](https://github.com/libretro/mame/assets/37503397/6353c4e1-3f8f-4b22-b3c7-b1ba8ab353b3)
![Screenshot_2023-08-10-18-54-19-714_com retroarch aarch64](https://github.com/libretro/mame/assets/37503397/aa8c4f58-e2dd-4ece-a445-db41576ca8d8)

**Comment by nemo93** [comment]
Thanks @Panderner. The devices we're running the various roms on are too low-spec to have games at fullspeed. Would you agree to close that ticket then? As there's no bug here. Also let me emphasize that some games do run lot faster on the FBNeo core.
